authentication/src/test/java/io/camunda/authentication/filters/SessionAuthenticationRefreshFilterTest.java

```diff
@@ -31,172 +32,241 @@
-      threads[i].start();
+      assertThat(testResult).hasStatusOk();
+      verify(authenticationProvider, times(1)).refresh();
+      final Instant lastRefreshTime = (Instant) session.getAttribute(LAST_REFRESH_ATTR);
+      assertThat(lastRefreshTime).isAfter(oldRefresh);
     }
 
-    for (final Thread thread : threads) {
-      thread.join();
+    @Test
+    void shouldOnlyRefreshOnceWhenMultipleConcurrentRequests() throws Exception {
+      final MockHttpSession session = new MockHttpSession();
+      setSessionRefreshAttribute(session, refreshInterval.multipliedBy(2));
+
+      final var result = getSendMultipleRequest(session, authenticationProvider, mockMvcTester);
+
+      assertThat(result.successfulRequests().get()).isEqualTo(result.threads().length);
+      verify(authenticationProvider, times(1)).refresh();
+      final Instant newRefreshTime = (Instant) session.getAttribute(LAST_REFRESH_ATTR);
+      assertThat(newRefreshTime).isAfter(Instant.now().minusMillis(refreshInterval.toMillis()));
+    }
+
+    private void setupAuthentication() {
+      final var mockAuthentication = createMockAuthentication();
+      final var testAuthentication = new TestingAuthenticationToken(mockAuthentication, null);
+      testAuthentication.setAuthenticated(true);
+      SecurityContextHolder.getContext().setAuthentication(testAuthentication);
+      when(authenticationProvider.getCamundaAuthentication()).thenReturn(mockAuthentication);
+    }
+
+    private CamundaAuthentication createMockAuthentication() {
+      return new CamundaAuthentication(
+          TestUserDetailsService.DEMO_USERNAME,
+          "",
+          Collections.emptyList(),
+          Collections.emptyList(),
+          Collections.emptyList(),
+          Collections.emptyList(),
+          new HashMap<>());
+    }
+
+    private void setSessionRefreshAttribute(
+        final MockHttpSession session, final Duration refreshInterval) {
+      final Instant lastRefreshRef = Instant.now().minus(refreshInterval);
+      session.setAttribute(LAST_REFRESH_ATTR, lastRefreshRef);
+      session.setAttribute(LAST_REFRESH_ATTR + "_LOCK", session.getId() + "LOCK");
     }
 
-    assertThat(successfulRequests.get()).isEqualTo(threads.length);
-    verify(authenticationProvider, times(1)).refresh();
-    final Instant newRefreshTime = (Instant) session.getAttribute(LAST_REFRESH_ATTR);
-    assertThat(newRefreshTime).isAfter(Instant.now().minusMillis(refreshInterval.toMillis()));
+    private SendMultipleRequest getSendMultipleRequest(
+        final MockHttpSession session,
+        final CamundaAuthenticationProvider authenticationProvider,
+        final MockMvcTester mockMvcTester)
+        throws InterruptedException {
+      doAnswer(
+              invocation -> {
+                Thread.sleep(100);
+                return null; // because it's a void method
+              })
+          .when(authenticationProvider)
+          .refresh();
```

Comment:
I could not detect a difference between doing nothing and this sleep block (that only evaluates once).
So I suggest we just use `doNothing()`, unless there is a specific reason not to.
```suggestion
      doNothing().when(authenticationProvider).refresh();
```

Reply:
OK, got it, ignore my comment then :)